torch/_functorch/aot_autograd.py

```diff
@@ -785,46 +822,36 @@ def maybe_to_fresh_input(idx, t):
             return t.view(t.shape)
         return t
 
-    def unpack_synthetic_bases(primals: List[Any]) -> List[Any]:
-        # This is only not None if our graph mutates a graph input that aliases another graph input.
-        if synthetic_base_info is None:
-            return primals
-
-        f_args_inner = []
-        for outer_idx_or_tuple in synthetic_base_info:
-            if isinstance(outer_idx_or_tuple, int):
-                f_args_inner.append(primals[outer_idx_or_tuple])
-            else:
-                outer_base_idx, view_tensor = outer_idx_or_tuple
-                outer_base = primals[outer_base_idx]
-                view_arg = gen_alias_from_base(
-                    outer_base, view_tensor, view_tensor.requires_grad
-                )
-                f_args_inner.append(view_arg)
-        return f_args_inner
+    def forward_or_joint(
```

Comment:
If possible, I suggest hoisting these helper functions to the top level, as it will make it clearer what variables they closed over versus not, and will make them more easily reusable if they are needed in other contexts.